src/MerQure.Tools/Buses/Bus.cs

```diff
@@ -0,0 +1,61 @@
+﻿using MerQure.Messages;
+using MerQure.Tools.Configurations;
+using MerQure.Tools.Messages;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace MerQure.Tools.Buses
+{
+    internal class Bus<T> : IBus<T> where T : IDelivered
+    {
+        internal Publisher<T> Producer { get; set; }
+        internal RetryConsumer<T> Consumer { get; set; }
+
+        public Bus(Publisher<T> producer, RetryConsumer<T> consumer)
+        {
+            Producer = producer;
+            Consumer = consumer;
+        }
+
+        public void AcknowlegdeDeliveredMessage(Channel channel, T deliveredMessage)
+        {
+            Consumer.AcknowlegdeDeliveredMessage(channel, deliveredMessage);
+        }
+
+        public void OnMessageReceived(Channel channel, EventHandler<T> callback)
+        {
+            Consumer.Consume(channel, callback);
+        }
+
+        public void Publish(Channel channel, T message, bool applyDeliveryDeplay = false)
+        {
+            Producer.Publish(channel, message, applyDeliveryDeplay);
+        }
+
+        public void PublishWithTransaction(Channel channel, IEnumerable<T> messages, bool applyDeliveryDeplay = false)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Rename parameter 'messages' to 'message' to match the interface declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS927)